editor/src/components/custom-code/code-file.ts

```diff
@@ -129,6 +130,10 @@ export interface ComponentDescriptor {
   preferredChildComponents: Array<PreferredChildComponentDescriptor>
   variants: ComponentInfo[]
   source: ComponentDescriptorSource
+  focus?: Focus
```

Comment:
why not `focus: Focus | undefined` here too?

Reply:
because this is the API type, and we want to allow the users to skip setting focus.